src/rgw/rgw_s3select.cc

```diff
@@ -485,6 +540,10 @@ int RGWSelectObj_ObjStore_S3::handle_aws_cli_parameters(std::string& sql_query)
   extract_by_tag(m_s3select_input, "QuoteCharacter", m_quot);
   extract_by_tag(m_s3select_input, "RecordDelimiter", m_row_delimiter);
   extract_by_tag(m_s3select_input, "FileHeaderInfo", m_header_info);
+  extract_by_tag(m_s3select_input, "Type", m_json_datatype);
```

Comment:
attached is the message delivered by the client-side (`AWS-CLI` --> `RGW`)

Content-Length: 363\r\n\r\n\<SelectObjectContentRequest xmlns="[http://s3.amazonaws.com/doc/2006-03-01/](http://s3.amazonaws.com/doc/2006-03-01/%5C)"\>\<Expression\>select count(0) from s3object[*];\</Expression\>\<ExpressionType\>SQL\</ExpressionType\>\<InputSerialization\>\<JSON\>\<Type\>DOCUMENT\</Type\>\</JSON\>\<CompressionType\>NONE\</CompressionType\>\</InputSerialization\>\<OutputSerialization\>\<CSV /\>\</OutputSerialization\>\</SelectObjectContentRequest\>"

the **InputSerialization** tag contains the object type (CSV / JSON / Parquet), the flow should use that.
the **Type** tag is a sub-tag (in the case of JSON it's "Document" or "Lines")